js/charts/clusteredhorizontalbar.js

```diff
@@ -885,75 +884,73 @@ function clusteredhorizontalbar() {
                 .on('mousemove', _handleMouseMoveFn.call(chart, tooltip, _local_svg))
                 .on('mouseout', _handleMouseOutFn.call(chart, tooltip, _local_svg))
                 .on('click', function (d) {
-                    if (COMMON.COMPARABLE_DATA_TYPES.indexOf(_dimensionType[0]) === -1) {
-                        if (!_print) {
-                            if (broadcast != undefined && broadcast.isThresholdAlert) {
-                                var ThresholdViz = {};
-                                ThresholdViz.ID = parentContainer.attr('vizID');
-                                ThresholdViz.measure = d.measure;
-                                ThresholdViz.measureValue = d[d.measure];
-                                ThresholdViz.dimension = d.dimension;
-                                ThresholdViz.dimensionValue = d[d.dimension];
-                                broadcast.ThresholdViz = ThresholdViz;
-                                broadcast.$broadcast('FlairBi:threshold-dialog');
+                    if (!_print) {
+                        if (broadcast != undefined && broadcast.isThresholdAlert) {
+                            var ThresholdViz = {};
+                            ThresholdViz.ID = parentContainer.attr('vizID');
+                            ThresholdViz.measure = d.measure;
+                            ThresholdViz.measureValue = d[d.measure];
+                            ThresholdViz.dimension = d.dimension;
+                            ThresholdViz.dimensionValue = d[d.dimension];
+                            broadcast.ThresholdViz = ThresholdViz;
+                            broadcast.$broadcast('FlairBi:threshold-dialog');
+                        }
+                        else {
+                            if (isLiveEnabled) {
+                                broadcast.$broadcast('FlairBi:livemode-dialog');
+                                return;
                             }
-                            else {
-                                if (isLiveEnabled) {
-                                    broadcast.$broadcast('FlairBi:livemode-dialog');
-                                    return;
-                                }
-                                filter = false;
-                                var confirm = parentContainer.select('.confirm')
-                                    .style('visibility', 'visible');
-                                var _filter = _Local_data.filter(function (d1) {
-                                    return d[_dimension[0]] === d1[_dimension[0]]
+                            filter = false;
+                            var confirm = parentContainer.select('.confirm')
+                                .style('visibility', 'visible');
+                            var _filter = _Local_data.filter(function (d1) {
```

Comment:
_filter is an array ..rename it to _filters

Reply:
yes its array but it always returns only one record